pkg/common/templates/templates.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package templates
+
+import (
+	"text/template"
+	"strings"
+	"github.com/golang/glog"
+  "sigs.k8s.io/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var kubeClient *kubernetes.Interface
+var kubeConfig *rest.Config
+var kubeAPIResourceList []*metav1.APIResourceList
+
+func InitializeKubeClient(kClient *kubernetes.Interface, kConfig *rest.Config) {
+	kubeClient = kClient
+	kubeConfig = kConfig
+}
+
+//If this is set, template processing will not try to rediscover
+// the apiresourcesList needed for dynamic client/ gvk look
+func SetAPIResources(apiresList []*metav1.APIResourceList) {
+	kubeAPIResourceList = apiresList
+}
+
+// just does a simple check for {{ string to indicate if it has a template
+func HasTemplate(templateStr string) bool{
+	glog.V(2).Infof("hasTemplate template str:  %v", templateStr)
+
+	hasTemplate := false
+	if strings.Contains(templateStr, "{{") {
+		hasTemplate = true
+	}
+
+	glog.V(2).Infof("hasTemplate: %v", hasTemplate)
+	return hasTemplate
+}
+
+// Main Template Processing func
+func ResolveTemplate(tmplMap interface{}) (interface{}, error) {
+
+  glog.V(2).Infof("ResolveTemplate for: %v", tmplMap)
+
+  // Build Map of supported template functions
+	funcMap := template.FuncMap{
+			"fromSecret": fromSecret,
+			"fromConfigMap": fromConfigMap,
+			"fromClusterClaim": fromClusterClaim,
+			"lookup" : lookup,
+			"base64enc" : base64encode,
+			"base64dec" : base64decode,
+	}
+
+  // create template processor and Initialize function map
+	tmpl := template.New("tmpl").Funcs(funcMap)
+
+
+	//convert the interface to yaml to string
+	// ext.raw is jsonMarshalled data which the template processor is not accepting
+	// so marshalling  unmarshalled(ext.raw) to yaml to string
+
+	templateStr := toYAML(tmplMap)
+
+	tmpl, err := tmpl.Parse(templateStr)
+	if err != nil {
+		glog.Errorf("error parsing the template %v", err)
+		return "", err
+	}
+
+	var buf strings.Builder
+	err = tmpl.Execute(&buf, "")
+	if err != nil {
+		glog.Errorf("error executing the template %v", err)
+	  return "", err
+	}
+
+	resolvedTemplateStr := buf.String()
+	glog.V(2).Infof("resolved template: %v ", resolvedTemplateStr)
```

Comment:
should this be cleaned up too?

Reply:
by default Info is not logged and it is also at level 2. .V(2).Infof . So this would appear in the logs only if log level is elevated to level 2 and  Info is enabled